docs/authorization-introduction.md

```diff
@@ -0,0 +1,96 @@
+- Provide PBAC authorization service for external applications.
+
+## Authorization elements
+
+The access control management evolved from the need to mitigate the risk of unauthorized access to data.
+
+Several control models were built to approach the various access control needs: Access Control Lists, Role Based Access Control, Attribute Based Access Control, Policy Based Access Control, Risk Adaptive Access Control.
+
+Udaru relies on [iam-js][] that is based on [node-pbac][] module, module that is inspired and mostly compatible with the [AWS IAM][] policy engine.
+
+A policy is a document that describes what type of access the policy owner has to a resource or a group of resources. The policy engine evaluates all the policies attached to a user (policies attached to the user, the teams(s) that the user belong to and the organization the user belongs to) and decides if the user has access to perform a certain action over a certain resource.
+
+For more details on policies and policy elements see [AWS IAM Policy][]
+
+A policy looks like:
+
+```javascript
+{
+  id: 'Policy ID',
+  version: '2016-07-01',
+  name: 'Policy name',
+  statements: { Statement: [{
+    Effect: 'Allow',
+    Action: ['dummy'],
+    Resource: ['dummy']
+  }] }
+}
+```
+
+The main elements of a policy are:
+- The policy ID,
+- The policy version,
+- The policy name,
+- The policy statements that contains an array of statements.
+
+The elements of a Statement are:
+- The Action to be performed. Example: `'authorization:teams:create'` or `'authorization:organizations:*'`,
+- The Resource on which the action is performed. Example: `'FOO:orga:CLOUDCUCKOO:scenario:*:entity:north-america-id'`,
+- The Effect - has the value 'Allow' or the value 'Deny'.
+
+Wildcards can be used on Action and Resource names.
+
+## Internal system authorization
+
+The Udaru authorization system uses the same PBAC approach to perform administration of its internal resources. More details can be seen on [Authorization Tech Spec][]
+
+The functionality endpoints are structured as the following:
+- [Private routes][] are routes that can be accessed only through a **Service Key** that is passed as a query string in the form `sig=<key>`. The **Service Key** is loaded by Udaru from the config file. These private routes provide functionality to create, update and delete policies,
+- [Public routes][] belong to two main functionality categories: one set of endpoints provide functionality to access and manage the Organizations, Teams, Users and Policies hierarchies. The other set of provided functionality is the actual access evaluation functionality. More details on access authorization are given on the next sections.
+
+## Routes in stand-alone server and Hapijs plugin modes
+
+When Udaru is used as stand-alone server or is used as hapi.js plugin it exposes its functionality through enpoints.
```

Comment:
how about `Each endpoint exposed by udaru (when used as a server or an hapi plugin) has an action and a resource associated with it. To perform a call to a certain endpoint, the caller[*] needs policies that allow him to do that specific action on that specific resource.` ?

[*] I think we need to add some doc on the fact that we are passing the caller id in the `Authorization` header, and also some doc on the `org` header